sentinelhub/geometry.py

```diff
@@ -159,9 +159,9 @@ def _tuple_from_list_or_tuple(
         :raises: TypeError
         """
         if len(bbox) == 4:
-            min_x, min_y, max_x, max_y = cast(Tuple[float, float, float, float], bbox)
+            min_x, min_y, max_x, max_y = bbox
         else:
-            (min_x, min_y), (max_x, max_y) = cast(Tuple[Tuple[float, float], Tuple[float, float]], bbox)
+            (min_x, min_y), (max_x, max_y) = bbox
         return float(min_x), float(min_y), float(max_x), float(max_y)
```

Comment:
Would it make sense to limit the various options how to define a bbox? it seems like a similar situation as in case of features in eolearn. But this is a separate issue, so this is just a comment.

Reply:
we already did quite a bit
the problem with that is that it is fairly code-breaking, so it's a sin of the past that we're slowly correcting